src/AssemblerReadGraph2.cpp

```diff
@@ -4,11 +4,143 @@ using namespace shasta;
 
 
 void Assembler::createReadGraph2(
-    uint32_t maxAlignmentCount,
-    uint32_t maxTrim)
+    uint32_t maxAlignmentCount)
 {
-    // This boilerplate code creates the read graph using
-    // all available alignments.
-    vector<bool> keepAlignment(alignmentData.size(), true);
+    // Percentile cutoffs for automated parameters
+    double largePercentile = 0.12;
+    double smallPercentile = 0.015;
+
+    vector<bool> keepAlignment(alignmentData.size(), false);
+
+    // Initialize histograms for measuring alignedFraction, markerCount, maxDrift, and maxSkip distributions
+    Histogram2 alignedFractionHistogram(0, 1, 100);
+    Histogram2 markerCountHistogram(0, 3000, 300);
```

Comment:
For long reads the maximum of 3000 might be insufficient. Consider modifying `Histogram2` so you no longer have to specify the maximum, and it adjusts the length of the vector as needed?